### PR TITLE
#3128 - PT Request a Change - Partner Information and Income (Student Form and Ministry Processing)

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Added new form:
image

- Added a new section form the Partner information and income appeal:
image

image

- Moved logic to use new calculated variable and CRA priority:
image

- Used new calculated variable:
image

- Added E2E tests

### CODE REVIEW

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn [hunk 1]
```diff
@@ -483,516 +519,423 @@ studentCalculatedTotalIncome</bpmn:text>
...
+      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">=calculatedDataTotalEligibleDependants &gt;= 3</bpmn:conditionExpression>
+    </bpmn:sequenceFlow>
+    <bpmn:sequenceFlow id="Flow_1lswqbt" name="Less than 3 dependants" sourceRef="Gateway_04hunux" targetRef="Event_1iix0z3">
+      <bpmn:conditionExpression xsi:type="bpmn:tFormalExpression">=calculatedDataTotalEligibleDependants &lt;= 2</bpmn:conditionExpression>
+    </bpmn:sequenceFlow>
+    <bpmn:sequenceFlow id="Flow_1p7fyev" sourceRef="Gateway_03c2fh6" targetRef="Activity_0pvobml" />
+    <bpmn:scriptTask id="Activity_01bf5sp" name="Partner Information and Income">
       <bpmn:extensionElements>
+        <zeebe:script expression="={&#10;  calculatedDataRelationshipStatus: appealsPartnerInformationAndIncomeAppealData.relationshipStatus,&#10;  calculatedDataPartner1TotalIncome: (if appealsPartnerInformationAndIncomeAppealData.relationshipStatus = &#34;married&#34; then appealsPartnerInformationAndIncomeAppealData.partnerEstimatedIncome else null)&#10;}" resultVariable="calculatedDataStudentPartnerInformationAndIncome" />
```
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn:842]: In the below logic, is the `if` required? Would it not be consistent with the selected form option?
```
{
  calculatedDataRelationshipStatus: appealsPartnerInformationAndIncomeAppealData.relationshipStatus,
  calculatedDataPartner1TotalIncome: (if appealsPartnerInformationAndIncomeAppealData.relationshipStatus = "married" then appealsPartnerInformationAndIncomeAppealData.partnerEstimatedIncome else null)
}
```

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn [hunk 2]
```diff
@@ -483,516 +519,423 @@ studentCalculatedTotalIncome</bpmn:text>
...
         </zeebe:ioMapping>
       </bpmn:extensionElements>
-      <bpmn:incoming>Flow_1fbmivh</bpmn:incoming>
-      <bpmn:outgoing>Flow_0n7won7</bpmn:outgoing>
+      <bpmn:incoming>Flow_0dqxcli</bpmn:incoming>
+      <bpmn:outgoing>Flow_0exl4fg</bpmn:outgoing>
     </bpmn:intermediateThrowEvent>
-    <bpmn:intermediateThrowEvent id="Event_00c1iyh" name="calculatedDataPartner1TotalIncome">
+    <bpmn:intermediateThrowEvent id="Event_0hxxmu9" name="Define Total Income">
```
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn:881]: Would it not be the case of moving the "Define Total Income" logic after the check for single/married, as below?
image

File: sources/packages/forms/src/form-definitions/partnerinformationandincomeappeal.json [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+      "encrypted": false,
+      "showCharCount": false,
+      "showWordCount": false,
+      "allowMultipleMasks": false,
+      "addons": [],
+      "id": "eamewph",
+      "hideLabel": true
+    },
+    {
+      "label": "<strong>On my first day of class, I'll be:</strong>",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/partnerinformationandincomeappeal.json:94]: Minor, removing the `<strong>` should not impact the UI. An existing CSS should be taken care of.

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn (hunk 2 above)
Comment (andrepestana-aot) [reply sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn:881]: I changed this way:
image
This way we don't even need to check CRA reported income when single. Thanks!

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-2022-2023.bpmn:845]: Yes, got it, but the check does not need to be based in the "relationshipStatus = "married", right?
The check based in the `relationshipStatus` would be a "business check" while the `is defined` with be a "technical check". I would prefer the last one for this appeals area. Does it make sense?

File: sources/packages/backend/workflow/test/2022-2023/parttime-assessment/appeals/parttime-assessment-partner-information-and-income.e2e-spec.ts
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/test/2022-2023/parttime-assessment/appeals/parttime-assessment-partner-information-and-income.e2e-spec.ts:1]: Amazing scenarios coverage ❤️

File: sources/packages/backend/workflow/test/models/assessment.model.ts [hunk 4]
```diff
@@ -33,6 +33,14 @@ export interface StudentAdditionalTransportationAppealData extends JSONDoc {
   additionalTransportPlacement?: YesNoOptions;
 }
 
+type relationshipStatusType = "single" | "other" | "married";
+
+export interface PartnerInformationAndIncomeAppealData extends JSONDoc {
+  partnerAuthorized?: boolean;
```
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/test/models/assessment.model.ts:39]: `partnerAuthorized` do we need this present as it is not part of the workflow?

File: sources/packages/backend/workflow/test/models/assessment.model.ts [hunk 5]
```diff
@@ -33,6 +33,14 @@ export interface StudentAdditionalTransportationAppealData extends JSONDoc {
   additionalTransportPlacement?: YesNoOptions;
 }
 
+type relationshipStatusType = "single" | "other" | "married";
```
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/test/models/assessment.model.ts:36]: Types should be declared as `RelationshipStatusType`.

Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/test/models/assessment.model.ts:36]: The type can be used for the E2Es assertions.

**Review by andrewsignori-aot — COMMENTED** [review]
Great work and really great to see the E2E coverage. Please take a look at the comments.

File: sources/packages/backend/workflow/test/models/assessment.model.ts (hunk 5 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/workflow/test/models/assessment.model.ts]: Should I consider that the suggestion was not accepted? Just double-checking, it was not a blocker.

**Review by andrewsignori-aot — APPROVED** [review]
Great work, looks good 👍

**Pull request merged**